### ISSUE TITLE
KeyError: 'x86_cr0'

### ISSUE THREAD

**Issue by ljun85** [angr/angr#173]
When use 'b.analyses.CFG()', it promt "KeyError: 'x86_cr0'", where b = angr.Project('/bin/true'), why?

```
File "/root/.virtualenvs/angr/local/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg/pyvex/block.py", line 70, in __init__
    1)
KeyError: 'x86_cr0'
```

**Comment by ltfish** [comment]
tl;dr The question has ben mentioned [here](https://github.com/angr/pyvex/issues/20). It looks like a version mismatch. Please make sure your `VEX`, `PyVEX`, and `archinfo` are all up-to-date.

Here are the details: `x86_cr0` is used at [line 17](https://github.com/angr/archinfo/blob/master/archinfo/arch_x86.py#L17) in `archinfo/archinfo/arch_x86.py`, and `vex_archinfo` is initialized [here](https://github.com/angr/archinfo/blob/master/archinfo/arch.py#L31). PyVEX has a C module, called `pyvex_c`, which provides this `vex_archinfo` thingy. My guess is, that somehow your VEX is not updated, or PyVEX is updated but not recompiled. You may recompile it by going into `/root/.virtualenvs/angr/local/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg/`, remove `pyvex_c.so`, and then do `pip install .`, which will recompile `pyvex_c.so` for you.

**Comment by ljun85** [comment]
Thank you very much. I will try it.

**Comment by ljun85** [comment]
To recompile PyVEX, i got into  /root/.virtualenvs/angr/local/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg/ ,  and , there are no  pyvex_c.so , is it the reason? 

After that, i did  pip install . ,  it prompted:
 "Requirement already satisfied (use --upgrade to upgrade): pycparser in /root/.virtualenvs/angr/lib/python2.7/site-packages (from pyvex==4.6.6.28)
Requirement already satisfied (use --upgrade to upgrade): cffi>=1.0.3 in /root/.virtualenvs/angr/lib/python2.7/site-packages/cffi-1.4.0-py2.7-linux-i686.egg (from pyvex==4.6.6.28)
Requirement already satisfied (use --upgrade to upgrade): archinfo in /root/.virtualenvs/angr/lib/python2.7/site-packages (from pyvex==4.6.6.28)
"

**Comment by ljun85** [comment]
I upgraded pycparser,cffi and archinfo, by pip install --upgrade, but it  prompted KeyError: 'x86_cr0' as before.

**Comment by ljun85** [comment]
Now, my VEX ,  PyVEX , and  archinfo  are all up-to-date,  but it prompted KeyError: 'x86_cr0' as before.

**Comment by ltfish** [comment]
Here is an example of `pyvex.default_vex_archinfo`:

```
In [1]: import pyvex

In [2]: pyvex.default_vex_archinfo()
Out[2]: 
<dictproxy {'arm64_dMinLine_lg2_szB': 0,
 'arm64_iMinLine_lg2_szB': 0,
 'endness': 1537,
 'hwcache_info': <dictproxy {'caches': None,
  'icaches_maintain_coherence': True,
  'num_caches': 0,
  'num_levels': 0}>,
 'hwcaps': 0,
 'ppc_dcbz_szB': 0,
 'ppc_dcbzl_szB': 0,
 'ppc_icache_line_szB': 0,
 'x86_cr0': 4294967295}>
```

If you are using the latest version of `PyVEX`, `x86_cr0` should be there. If your output is different from the above, it's likely that you are not using the latest `PyVEX`. Can you please check if `pyvex.__file__` is pointing to where it should be (the latest version)? You may also try to remove and reinstall `PyVEX` by doing `pip uninstall pyvex && pip install pyvex`, and see if it helps.

**Comment by ljun85** [comment]
Thanks for your anwser.  I did as you said.
In[1]:
         import pyvex
In[2]:
         pyvex.default_vex_archinfo()
Out[2]:
          {'arm64_dMinLine_lg2_szB': 0, 
          'hwcaps': 0,
           'ppc_icache_line_szB': 0,
           'ppc_dcbzl_szB': 0, 
           'endness': 1537, 'ppc_dcbz_szB': 0, 
           'arm64_iMinLine_lg2_szB': 0, 
           'x86_cr0': 4294967295L, 
           'hwcache_info': {'num_caches': 0, 'num_levels': 0, 'icaches_maintain_coherence': True, 'caches':    None}
            }

In[3]: 
         pyvex.__file__
Out[3]:
           /root/.virtualenvs/angr_3/local/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg/pyvex/__init__.pyc

But, when I tried to remove and reinstall  PyVEX  by doing  pip uninstall pyvex && pip install pyvex ,it promted that the latest version of PyVEX is "pyvex-4.6.6.28",  it was as same as just uninstall. details:
In[1]: 
         pip uninstall pyvex
Out[1]:
        Uninstalling pyvex:
         /root/.virtualenvs/angr_3/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg
        Proceed (y/n)? y
        Successfully uninstalled pyvex
In[2]:
        pip install pyvex
Out[2]:
        Downloading/unpacking pyvex
         Downloading pyvex-4.6.6.28.tar.gz (5.5Mb):   2%  118Kb
Is it meaning that the latest version of pyvex is "pyvex-4.6.6.28.tar.gz "?  it seems to contradicts your point.

**Comment by ljun85** [comment]
I  update my angr by donging  pip install --upgrade angr:
out:
Downloading/unpacking angr
  Downloading angr-4.6.6.29.tar.gz (206Kb): 206Kb downloaded
  Running setup.py egg_info for package angr

Requirement already up-to-date: capstone in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/capstone-3.0.4-py2.7.egg (from angr)
Requirement already up-to-date: networkx in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/networkx-1.11-py2.7.egg (from angr)
Requirement already up-to-date: futures in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/futures-3.0.5-py2.7.egg (from angr)
Cannot fetch index base URL http://pypi.python.org/simple/
Requirement already up-to-date: progressbar in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/progressbar-2.3-py2.7.egg (from angr)
Requirement already up-to-date: mulpyplexer in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from angr)
Requirement already up-to-date: cooldict in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from angr)
Requirement already up-to-date: ana in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from angr)
Requirement already up-to-date: archinfo in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/archinfo-4.6.6.28-py2.7.egg (from angr)
Requirement already up-to-date: pyvex in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg (from angr)
Requirement already up-to-date: claripy in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/claripy-4.6.6.29-py2.7.egg (from angr)
Requirement already up-to-date: simuvex in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/simuvex-4.6.6.28-py2.7.egg (from angr)
Requirement already up-to-date: cle in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/cle-4.6.6.28-py2.7.egg (from angr)
Requirement already up-to-date: cachetools in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/cachetools-1.1.6-py2.7.egg (from angr)
Requirement already up-to-date: decorator>=3.4.0 in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/decorator-4.0.10-py2.7.egg (from networkx->angr)
Requirement already up-to-date: pyelftools in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from archinfo->angr)
Requirement already up-to-date: pycparser in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/pycparser-2.14-py2.7.egg (from pyvex->angr)
Requirement already up-to-date: cffi>=1.0.3 in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/cffi-1.7.0-py2.7-linux-i686.egg (from pyvex->angr)
Requirement already up-to-date: angr-only-z3-custom in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/angr_only_z3_custom-4.4.2.alpha1-py2.7.egg (from claripy->angr)
Requirement already up-to-date: bintrees in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/bintrees-2.0.4-py2.7.egg (from simuvex->angr)
Requirement already up-to-date: dpkt-fix in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from simuvex->angr)
Requirement already up-to-date: idalink in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from cle->angr)
Cannot fetch index base URL http://pypi.python.org/simple/
Requirement already up-to-date: pefile in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/pefile-2016.3.28-py2.7.egg (from cle->angr)
Could not find any downloads that satisfy the requirement future in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/future-0.15.2-py2.7.egg (from pefile->cle->angr)
No distributions at all found for future in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/future-0.15.2-py2.7.egg (from pefile->cle->angr)
Storing complete log in /root/.pip/pip.log
(angr_4)root@ubuntu:/opt/ljun/angr/angr/angr-4.6.1.12# pip install --upgrade angr
Downloading/unpacking angr
  Running setup.py egg_info for package angr

Requirement already up-to-date: capstone in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/capstone-3.0.4-py2.7.egg (from angr)
Requirement already up-to-date: networkx in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/networkx-1.11-py2.7.egg (from angr)
Requirement already up-to-date: futures in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/futures-3.0.5-py2.7.egg (from angr)
Cannot fetch index base URL http://pypi.python.org/simple/
Requirement already up-to-date: progressbar in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/progressbar-2.3-py2.7.egg (from angr)
Requirement already up-to-date: mulpyplexer in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from angr)
Requirement already up-to-date: cooldict in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from angr)
Requirement already up-to-date: ana in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from angr)
Requirement already up-to-date: archinfo in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/archinfo-4.6.6.28-py2.7.egg (from angr)
Cannot fetch index base URL http://pypi.python.org/simple/
Requirement already up-to-date: pyvex in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg (from angr)
Requirement already up-to-date: claripy in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/claripy-4.6.6.29-py2.7.egg (from angr)
Requirement already up-to-date: simuvex in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/simuvex-4.6.6.28-py2.7.egg (from angr)
Requirement already up-to-date: cle in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/cle-4.6.6.28-py2.7.egg (from angr)
Cannot fetch index base URL http://pypi.python.org/simple/
Requirement already up-to-date: cachetools in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/cachetools-1.1.6-py2.7.egg (from angr)
Requirement already up-to-date: decorator>=3.4.0 in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/decorator-4.0.10-py2.7.egg (from networkx->angr)
Requirement already up-to-date: pyelftools in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from archinfo->angr)
Requirement already up-to-date: pycparser in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/pycparser-2.14-py2.7.egg (from pyvex->angr)
Requirement already up-to-date: cffi>=1.0.3 in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/cffi-1.7.0-py2.7-linux-i686.egg (from pyvex->angr)
Requirement already up-to-date: angr-only-z3-custom in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/angr_only_z3_custom-4.4.2.alpha1-py2.7.egg (from claripy->angr)
Requirement already up-to-date: bintrees in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/bintrees-2.0.4-py2.7.egg (from simuvex->angr)
Requirement already up-to-date: dpkt-fix in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from simuvex->angr)
Requirement already up-to-date: idalink in /root/.virtualenvs/angr_4/lib/python2.7/site-packages (from cle->angr)
Requirement already up-to-date: pefile in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/pefile-2016.3.28-py2.7.egg (from cle->angr)
Requirement already up-to-date: future in /root/.virtualenvs/angr_4/lib/python2.7/site-packages/future-0.15.2-py2.7.egg (from pefile->cle->angr)
Installing collected packages: angr
  Running setup.py install for angr

Successfully installed angr
Cleaning up...



Now, my angr compents are all up-to-date. But , i meat the "KeyError: 'x86_cr0'"  when i imput " b = angr.Project('/bin/true') ". details:
In[1]:
import angr
In[2]:
b = angr.Project('/bin/true')
Out[2]:
b = angr.Project('/bin/true')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/project.py", line 164, in __init__
    self._simos.configure_project()
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/simos.py", line 386, in configure_project
    super(SimLinux, self).configure_project()
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/simos.py", line 172, in configure_project
    self.proj.loader.perform_irelative_relocs(irelative_resolver)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/cle-4.6.6.28-py2.7.egg/cle/loader.py", line 799, in perform_irelative_relocs
    val = resolver_func(resolver)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/simos.py", line 161, in irelative_resolver
    val = resolver()
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/surveyors/caller.py", line 50, in __call__
    self.perform_call(*args)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/surveyors/caller.py", line 70, in perform_call
    caller_end_unpruned = caller.step(until=lambda pg: len(pg.active) == 0, step_func=step_func if self._concrete_only else None).unstash(from_stash='deadended')
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/path_group.py", line 515, in step
    pg = pg._one_step(stash=stash, selector_func=selector_func, successor_func=successor_func, check_func=check_func, **kwargs)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/path_group.py", line 306, in _one_step
    r = self._one_path_step(a, successor_func=successor_func, check_func=check_func, **kwargs)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/path_group.py", line 207, in _one_path_step
    successors = a.step(**kwargs)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/path.py", line 317, in step
    self._make_sim_run(throw=throw)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/path.py", line 343, in _make_sim_run
    self._run = self._project.factory.sim_run(self.state, **self._run_args)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/factory.py", line 133, in sim_run
    r = self.sim_block(state, addr=addr, **block_opts)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/factory.py", line 72, in sim_block
    **block_opts)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/angr/lifter.py", line 133, in lift
    traceflags=traceflags)
  File "/root/.virtualenvs/angr_4/local/lib/python2.7/site-packages/pyvex-4.6.6.28-py2.7.egg/pyvex/block.py", line 70, in __init__
    1)
KeyError: 'x86_cr0'

**Comment by rhelmot** [comment]
I'm not sure what to say at this point. There's clearly something very badly fucked up with your python environment. The only thing I can think of is to make sure no angr components (angr, simuvex, claripy, cle, pyvex, archinfo) are installed outside your virtualenv, and then try installing angr again in a new virtualenv.

**Comment by ljun85** [comment]
I will try it agian.
My steps is:
step1: mkvirtualenv angr_4
step2:pip install angr

**Comment by rhelmot** [comment]
I recommend you run `pip uninstall angr simuvex claripy cle pyvex archinfo` OUTSIDE your virtualenv first, perhaps several times, to make sure nothing is installed globally or in your user-local env.

**Comment by ljun85** [comment]
Thank you very much.

**Comment by ljun85** [comment]
Could i run "rmvirtualenv angr_4 " to make sure nothing is installed globally or in my user-local env.

**Comment by ljun85** [comment]
Unfortunately, i can't run  pip uninstall angr simuvex claripy cle pyvex archinfo  OUTSIDE my virtualenv , it promt my pip command has'n installed.

**Comment by rhelmot** [comment]
That's pretty terrifying? ok, we can try this:

- First, run `rmvirtualenv angr_4`
- Next, start up a python interpreter and run `import archinfo`. If it does not raise an exception, abort this sequence of instructions and tell me this
- Then, run `mkvirtualenv angr_4`
- Then, run `pip install angr`

**Comment by ljun85** [comment]
In[1]:
rmvirtualenv angr_4
In[2]:
python
In[3]:
import archinfo
Out[3]:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named archinfo

**Comment by rhelmot** [comment]
Ok, good, looks good. You can recreate the virtualenv and reinstall angr now. If it still fails I'm at a loss.

**Comment by ljun85** [comment]
Thanks at all.

**Comment by ljun85** [comment]
I run pip install angr, then , it prompted error as bellow :
 Running setup.py install for angr-only-z3-custom
    WARNING: '' not a valid package name; please use only.-separated package names in setup.py
    Configuring Z3
    error: Permission denied
    Complete output from command /root/.virtualenvs/angr_4/bin/python -c "import setuptools;__file__='/root/.virtualenvs/angr_4/build/angr-only-z3-custom/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-version-externally-managed --record /tmp/pip-hSVkMw-record/install-record.txt --install-headers /root/.virtualenvs/angr_4/include/site/python2.7:
    WARNING: '' not a valid package name; please use only.-separated package names in setup.py

running install

running build

Configuring Z3

error: Permission denied

**Comment by rhelmot** [comment]
- What operating system are you running?
- Are you using the root user?
- If not, why does it look like you are? (filepaths are in root's homedir)
- If yes, why are you getting permission denied?

**Comment by ljun85** [comment]
My operating system is Ubuntu 12.04.4 LTS.
Yes , i use the root user.

**Comment by rhelmot** [comment]
Oh.

For various technical reasons about the python install process we can't support ubuntu 12. You need to be using at least 14.04 LTS.

Try this:

```
pip install -U pip
pip install angr
```

If that doesn't work, I have no way of helping you besides telling you to upgrade your operating system :(

**Comment by ljun85** [comment]
Ok.

**Comment by ljun85** [comment]
I run pip install -U pip , and pip install angr, it prompted as bellow:
InsecurePlatformWarning
  Could not fetch URL https://pypi.python.org/simple/angr-only-z3-custom/: There was a problem confirming the ssl certificate: [Errno 1] _ssl.c:504: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed - skipping

Is there anything i can do except upgrading my operating system ?

**Comment by rhelmot** [comment]
Like, if you're really really good at linux stuff there's probably a way for you to get to the bottom of this and fix it? But it's completely impossible for me to guide you through fixing what is at this point a real serious problem with your python environment.

Sorry :(

**Comment by ljun85** [comment]
Thanks, i get it.

**Comment by ltfish** [comment]
> I run pip install -U pip , and pip install angr, it prompted as bellow:
> InsecurePlatformWarning
> Could not fetch URL https://pypi.python.org/simple/angr-only-z3-custom/: There was a problem confirming the ssl certificate: [Errno 1] _ssl.c:504: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed - skipping

Looks like an HTTPS verification issue. I would guess it is because you are using an old Ubuntu, and its local certificate chain (I don't remember the exact term, but whatever) needs to be updated. As a workaround, try running `pip install --index-url=http://pypi.python.org/simple/ --trusted-host pypi.python.org angr` in your virtualenv.

Source: http://stackoverflow.com/questions/25981703/pip-install-fails-with-connection-error-ssl-certificate-verify-failed-certi

**Comment by ljun85** [comment]
Thanks for your answer. I update my oprating system as Ubuntu 12.04.4 LTS. Then, I run pip install angr, and got the z3 problem. Details as bellow:
In[1]:cat /etc/issue
Out[1]:Ubuntu 12.04.4 LTS \n \l
In[2]:(angr_4)root@ubuntu:~/.virtualenvs# pip install angr
out[2]:
Installing collected packages: angr-only-z3-custom, bintrees, dpkt-fix, idalink, pefile, rpyc,                 future, plumbum
  Running setup.py install for angr-only-z3-custom
    WARNING: '' not a valid package name; please use only.-separated package names in setup.py
    Configuring Z3
    error: Permission denied
    Complete output from command /root/.virtualenvs/angr_4/bin/python -c "import setuptools;__                file__='/root/.virtualenvs/angr_4/build/angr-only-z3-custom/setup.py';exec(compile(open(__file                __).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-version-externally-mana                ged --record /tmp/pip-aahAd3-record/install-record.txt --install-headers /root/.virtualenvs/an                gr_4/include/site/python2.7:
    WARNING: '' not a valid package name; please use only.-separated package names in setup.py

running install

running build

Configuring Z3

error: Permission denied

Where, the angr_4 is the only on virtualenv. What can i do now? Thanks.

**Comment by ljun85** [comment]
As you see, i use the root user.

**Comment by ljun85** [comment]
Thank you very much, i will try .

**Comment by ljun85** [comment]
I upgrade my oprating system as "Ubuntu 14.04.5 ", then i runed apt-get install python-dev libffi-dev build-essential virtualenvwrapper , and maked the only one virtualenv "angr_env", then i run pip install angr. Unfortunately, i got the z3 issue as before. Details as bellow:
In[1]:cat /etc/issue
Out[1]:Ubuntu 14.04.5 LTS \n \l
In[2]:(angr_env)root@ubuntu:/# pip install angr
Out[2]:
.......
 Configuring Z3
    error: [Errno 13] Permission denied
    Complete output from command /root/.virtualenvs/angr_env/bin/python -c "import setuptools;__file__='/root/.virtualenvs/angr_env/build/angr-only-z3-custom/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Yegzx7-record/install-record.txt --single-version-externally-managed --install-headers /root/.virtualenvs/angr_env/include/site/python2.7:
    WARNING: '' not a valid package name; please use only.-separated package names in setup.py


As you see, i used root user.

**Comment by ljun85** [comment]
My Python version is "Python 2.7.6"

**Comment by ljun85** [comment]
I  download  angr-only-z3-custom, and installed  it manually, then i maked it. Thanks at all.

**Comment by ljun85** [comment]
I  download  angr-only-z3-custom, and installed  it manually, then i maked it. Thanks at all.

**Comment by ljun85** [comment]
I  download  angr-only-z3-custom, and installed  it manually, then i maked it. Thanks at all.

**Comment by ljun85** [comment]
I  download  angr-only-z3-custom, and installed  it manually, then i maked it. Thanks at all.

**Comment by ljun85** [comment]
I  download  angr-only-z3-custom, and installed  it manually, then i maked it. Thanks at all.

**Comment by ljun85** [comment]
I  download  angr-only-z3-custom, and installed  it manually, then i maked it. Thanks at all.

**Comment by csdrane** [comment]
Running into the same Z3 issue building with Debian 7.9.

**Comment by csdrane** [comment]
I was able to successfully pip install angr by manually installing angr-only-z3-custom as well. Also needed to manually apt-get install capstone.

**Comment by ltfish** [comment]
@csdrane Thanks for sharing your solution!
